processing/src/main/java/org/apache/druid/segment/QueryableIndexIndexableAdapter.java

```diff
@@ -113,6 +111,13 @@ public <T extends Comparable<? super T>> CloseableIndexed<T> getDimValueLookup(S
     final BaseColumn col = columnHolder.getColumn();
 
     if (!(col instanceof DictionaryEncodedColumn)) {
+      // this shouldn't happen, but if it does, try to close to prevent a leak
+      try {
+        col.close();
+      }
+      catch (IOException e) {
+        throw new RuntimeException(e);
+      }
```

Comment:
A bit of a nit, but if you add this text to the exception message, you get the benefit of the comment explaining that you don't expect it to occur and if the exception ever does get thrown, the message is there too for whoever sees it.